### ISSUE TITLE
How to request email after login ?

### ISSUE THREAD

**Issue by V-Klimov** [playgameservices/play-games-plugin-for-unity#2466]
Hello.

In the old version of GPG, the game could request a user email when needed. 
In the new - must request permission at login (via builder.RequestEmail).

This is extremely bad because the user does not understand why he should give an email. 

How can the game get an email in the old style, requesting permission after login?

**Comment by olehkuznetsov** [comment]
Probably you can achieve the same if you use 
```
           PlayGamesClientConfiguration ClientConfiguration =
                new PlayGamesClientConfiguration.Builder()
                     .RequestEmail()
                       ... other options were used before
                     .Build();
            PlayGamesPlatform.InitializeInstance(ClientConfiguration);
            PlayGamesPlatform.Activate();
            Social.localUser.Authenticate(...);
```
if you need the email later.

**Comment by V-Klimov** [comment]
> Probably you can achieve the same if you use
> 
> ```
>            PlayGamesClientConfiguration ClientConfiguration =
>                 new PlayGamesClientConfiguration.Builder()
>                      .RequestEmail()
>                        ... other options were used before
>                      .Build();
>             PlayGamesPlatform.InitializeInstance(ClientConfiguration);
>             PlayGamesPlatform.Activate();
>             Social.localUser.Authenticate(...);
> ```
> if you need the email later.

No. In this case, the game asks for permission to access email at the login.

**Comment by olehkuznetsov** [comment]
First time login do not add .RequestEmail()
You can call this configuration/authentication code later again with '.RequestEmail' if you need email.

**Comment by V-Klimov** [comment]
Can you provide a code sample?
Something like this:
`
PlayGamesClientConfiguration.Builder builder;

void login ( ){
  builder = new PlayGamesClientConfiguration.Builder();
  PlayGamesClientConfiguration config 	= builder.Build();
  PlayGamesPlatform.InitializeInstance	( config ); 
  PlayGamesPlatform.Activate	( );
  Social.localUser.Authenticate ( auth_callback );
}

string request_email ( ){
  builder.RequestEmail( );
  return PlayGamesPlatform.Instance.GetUserEmail ( );
}
`

??

**Comment by olehkuznetsov** [comment]
request_email ( callback ) {
  builder.RequestEmail( );
  PlayGamesClientConfiguration config 	= builder.Build();
  PlayGamesPlatform.InitializeInstance	( config ); 
  PlayGamesPlatform.Activate	( );
  Social.localUser.Authenticate ( authenticated => { callback(player.getEmail()); } );
}

**Comment by V-Klimov** [comment]
thanks, I'll try

**Comment by V-Klimov** [comment]
> ```
> void request_email ( callback ) {
>   builder.RequestEmail( );
>   PlayGamesClientConfiguration config 	= builder.Build();
>   PlayGamesPlatform.InitializeInstance	( config ); 
>   PlayGamesPlatform.Activate	( );
>   Social.localUser.Authenticate ( authenticated => { callback(player.getEmail()); } );
> }
> ```

This code useless. It's impossible to initialize instance twice. Check PlayGamesPlatform.cs:

```
        public static void InitializeInstance(PlayGamesClientConfiguration configuration)
        {
            if (sInstance != null)
            {
                GooglePlayGames.OurUtils.Logger.w(
                    "PlayGamesPlatform already initialized. Ignoring this call.");
                return;
            }

            sInstance = new PlayGamesPlatform(configuration);
        }
```

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
